### PR TITLE
[FEATURE][86drmx95g] Send email when someone subscribes to EO

### PR DESCRIPTION
**Pull request by Mat0ias**
# Light It Pull Request Template ⚡⚡⚡

_Resolve_: [closes -Slack notification when someone subscribes to the eo email panel](https://app.clickup.com/t/86drmx95g)

## Description and Requires ⭐

- Post **email** to the back-end when someones subscribes to eo
- Disable button when there is not a valid email

## Proof of Test (optional) 📷

https://github.com/Light-it-labs/eo_web/assets/132606537/2bb82f72-3a4f-4437-a3c9-07c755584780

## Checklist ✅

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/eo_web/src/components/EOInYourInbox.tsx [hunk 1]
```diff
@@ -1,8 +1,26 @@
...
+
+    try {
+      await SubscribeToEoEmailPost(email);
+    } catch (error) {
+      console.error('Error submitting email:', error);
+    } finally {
+      setIsLoading(false)
+      setSubmitted(true)
+    }
+  };
```
[the remark this answers is not in the document]
Comment (Mat0ias) [reply apps/eo_web/src/components/EOInYourInbox.tsx:22]: Done!

File: packages/ui/src/common/Button.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ export const Button = forwardRef(
       ref={ref}
       type={type}
       className={tw(
-        "flex h-12 flex-row items-center justify-between gap-2 border border-transparent text-center focus:outline-none",
+        "flex h-12 flex-row items-center justify-between gap-2 border border-transparent text-center focus:outline-none disabled:cursor-not-allowed",
```
[the remark this answers is not in the document]
Comment (Mat0ias) [reply packages/ui/src/common/Button.tsx:51]: Done!

File: apps/eo_web/src/components/EOInYourInbox.tsx [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
+import { toast } from 'react-toastify'
 
 export const EOInYourInbox = () => {
   const [submitted, setSubmitted] = useState(false)
+  const [email, setEmail] = useState('')
+
+  const { mutate, isLoading } = useMutation({
+    mutationFn: SubscribeToEoEmailPost,
+    onSuccess: () => setSubmitted(true),
+    onError: () => toast.error("Something went wrong, try again.")
```
Comment (sgoycoechea-lightit) [inline apps/eo_web/src/components/EOInYourInbox.tsx:14]:
```suggestion
    onError: () => toast.error("Something went wrong")
```

**Review by sgoycoechea-lightit — APPROVED** [review]

File: apps/eo_web/src/api/email.ts [hunk 4]
```diff
@@ -10,3 +10,14 @@ export const resendEmailVerification = async (email: string) => {
     { email },
   );
 };
+
+export interface SubscribeToEoEmailPostResponse {
+  success: boolean;
+}
+
+export const SubscribeToEoEmailPost = async (email: string) => {
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/api/email.ts:18]: Rename this function to start with lower case

**Review by sgoycoechea-lightit — APPROVED** [review]

**Pull request merged**